### ISSUE TITLE
use as many connections as URLs

### ISSUE THREAD

**Issue by ordex** [eribertomota/axel#92]
If more than one URL is provided by the user, axel should use at least as many connections as the number of URLs. If not, some URLs would go unused.

Maybe we should rather set num-connections as (num-connections x num of URLs) ?

**Comment by ismaell** [comment]
Perhaps when the flag isn't specified... definitely not num_urls*num_connections.

**Comment by ordex** [comment]
yeah, that could be a good idea. if no flag is specified, then we spawn as many connections as the urls.
If the flag is specified but the number is less than the urls then we print a warning

**Comment by ismaell** [comment]
With some reasonable limit. I think it's not a bad idea to specify more URLs than connections given it could be just for fallback (do we handle that correctly?).

**Comment by ordex** [comment]
I am not sure if we do handle that correctly. we should check that
